src/backend/test.rs

```diff
@@ -916,4 +986,786 @@ mod tests {
         let mut backend = TestBackend::new(10, 2);
         backend.flush().unwrap();
     }
+
+    #[test]
+    #[cfg(feature = "scrolling-regions")]
+    fn scroll_region_up_whole_screen_by_0() {
+        let mut backend = TestBackend::new(10, 5);
+        backend.buffer = Buffer::with_lines([
+            "aaaaaaaaaa",
+            "bbbbbbbbbb",
+            "cccccccccc",
+            "dddddddddd",
+            "eeeeeeeeee",
+        ]);
```

Comment:
I normally avoid DRYing tests too much, but I notice that the visual height of the tests seems to get in the way of what they're testing at a glance. The concepts the tests are testing don't really have much to do with the display, but more to do with the idea that the buffer is unchanged. This might make each of the lines seem better as a constant.

Perhaps something like
```rust
// either
const LINE_1: &str = "aaaaaaaaaa";
...
// or
const LINES: [&str; 5] = [...]
const EMPTY_LINE: ...
```
and then:
```rust
backend.assert_buffer_lines([LINE_1, LINE_2, LINE_3, LINE_4, LINE_5]);
```

Alternatively, considering the line length doesn't matter too much (only that there's some width > 1), perhaps collapse the lines to just "aa" etc. and the assertions may naturally wrap at a single line.


Edit: actually, this might be even better purely as an rstest:

```rust
#[rstest]
#[case::whole_screen_by_0(0..5, 0, &[], &[A, B, C, D, E])
#[case::whole_screen_by_2(0..5, 2, &[A, B], &[C, D, E])
...
fn scroll_up(#[case]line_range: Range<u16> #[case]line_count: u16, #[case]expected_scrollback: &[&str], #[case]expected_buffer: &[&str]) {
    let backend = ...
    let backend.buffer = Buffer::with_lines([A,B,C,D,E);
    backend.scroll_region_up(line_range, line_count).unwrap();
    backend_assert_scrollback(expected_scrollback);
    backend_assert_buffer_lines(expected_buffer);
}
```

Reply:
Okay, I did this. Let me know what you think.